### ISSUE TITLE
Replace comments with spaces instead of stripping

### ISSUE THREAD

**Issue by bcowgill** [json5/json5#119]
So that error messages from other tools which parse the json will match up as close as possible to the source json5

**Comment by jordanbtucker** [comment]
I'm not sure I follow. If your JSON5 file is invalid, then you'll get errors when you try to compile it to JSON. A JSON file generated from a JSON5 source file would not have any errors.

If you're talking about application errors not related to JSON syntax (e.g. the JSON consuming application expected a number instead of a string) can you provide an example where replacing comments with a space instead of stripping them would match up more closely with the source file?

**Comment by bcowgill** [comment]
I mean what you said in your second paragraph. I don't have a real world example yet, as far as JSON is concerned. As in, this is not stopping me. I have felt this pain before using other source code translators though. As you say, if a consuming app reported an error as line 5 column 32
then the removal of commented lines would throw the line number off and any inline comments
would throw the column count off.

While looking for a commented JSON solution I saw this one: 

https://github.com/sindresorhus/strip-json-comments

Which mentions that feature as an option and thought I'd post it here as well.

BTW json5 worked just perfectly for my project yesterday so thanks for that!

**Comment by jordanbtucker** [comment]
Okay, it makes more sense after reading the README for strip-json-comments.

Comments are replaced with whitespace, meaning spaces and new lines where appropriate, to keep the lines and columns matching. Here's some examples:

```js
{
  // happy
  "earth": "day"
}

{
  
  "earth": "day"
}
```

```js
{
  "earth-day-2016": /* 4/22/2016 in ms */ 1461308400000
}

{
  "earth-day-2016":                       1461308400000
}
```

```js
{
  "first-earth-day":
  /* Earth Day was founded by
     Senator Gaylord Nelson on
     April 22, 1970 */
  "1970-04-22T00:00:00.000Z"
}

{
  "first-earth-day":
  
  
  
  "1970-04-22T00:00:00.000Z"
}
```

**Comment by jordanbtucker** [comment]
I don't think this makes sense for this project since JSON5 to JSON is not a one-to-one compilation even without comments. JSON5 documents are normalized to JSON as if you used the `JSON.stringify` function.

Implementing [sourcemaps](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit?hl=en_US&pli=1&pli=1) might be a better general option.

**Comment by bcowgill** [comment]
I realised it also wouldn't help in the case where you split lines using \ as the following line numbers would also be out of alignment.
